providers/redhat/feed.go

```diff
@@ -53,7 +55,10 @@ func loadFeed(r io.Reader) (*Feed, error) {
 }
 
 // Checker returns an rpm.Checker that uses the Feed.
-func (feed Feed) Checker() (rpm.Checker, error) {
+func (feed *Feed) Checker() (rpm.Checker, error) {
```

Comment:
Probably a good idea to check `feed` for `nil`, since you changed it to a pointer... Just in case, I know it's always called on an instance in the code now. A nitpick, tho.

Reply:
The method works with zero values, that's already a lot :)

I don't think we should encourage the pattern of checking for nil pointer receivers, having extra error paths all over a code base vs panicking on nil receivers from an incorrect usage of the API, I'd rather we opt for the second.

As an anecdotal evidence, I don't remember having seen any go code base checking for nil receivers.

Reply:
I did not mean returning an error necessarily as much as panic'ing with some meaningful message, rather than "runtime error: invalid memory address or nil pointer dereference". But yeah, not a biggie.

Reply:
Thanks for merging this!